### PR TITLE
[MRG] FIX: NumPy errors in advanced examples

### PR DESCRIPTION
**Pull request by KamalakerDadi**
From PR #1378 it seems like there are some examples missed to change.
See https://circleci.com/gh/nilearn/nilearn/2263

This PR would hopefully fix failures related to recent NumPy.

Any reviews please ?

### CODE REVIEW

**Comment by bthirion** [comment]
LGTM

File: examples/01_plotting/plot_haxby_masks.py [hunk 1]
```diff
@@ -27,7 +27,7 @@
 affine = mean_img.affine
 _, _, k_slice = coord_transform(0, 0, z_slice,
                                 linalg.inv(affine))
-k_slice = np.round(k_slice)
+k_slice = int(np.round(k_slice))
```
Comment (GaelVaroquaux) [inline examples/01_plotting/plot_haxby_masks.py]: Do we still need the np.round here?

Comment (KamalakerDadi) [inline examples/01_plotting/plot_haxby_masks.py]: Yes. For instance, k_slice=array(27.766666666666666).

Comment (KamalakerDadi) [inline examples/01_plotting/plot_haxby_masks.py]: May be not. Shall I remove ?

**Comment by KamalakerDadi** [comment]
I got some issues with affine in this examples. I am fixing them too.

File: examples/01_plotting/plot_haxby_masks.py (hunk 1 above)
Comment (GaelVaroquaux) [inline examples/01_plotting/plot_haxby_masks.py]: Let's leave it

**Comment by GaelVaroquaux** [comment]
Are you saying that you need to fix extra stuff and that I should wait before merging?

**Comment by KamalakerDadi** [comment]
>Are you saying that you need to fix extra stuff and that I should wait before merging?

Yes, please wait. In the examples, the way we get ```affine``` looks like a failure. Can you have a quick look ? In all examples.

**Comment by KamalakerDadi** [comment]
>In all examples. 

I meant, all examples with this PR.

**Comment by KamalakerDadi** [comment]
I made an attempt to fix some failures. Can I have reviews please ?

**Comment by GaelVaroquaux** [comment]
@KamalakerDadi : it seems to me that in some of the examples this whole code jumping from slice index to z coordinate in MNI space a bit pointless (and ugly IMHO). We are hard-coding a slice position, we could as well do it with a hard coded position if we don't use the slice position.

If I am not wrong, in examples/01_plotting/plot_haxby_masks.py we could just remove that part of the code. I'll merge this, and do it myself, it's easy.

Thanks! Merging!

**Comment by GaelVaroquaux** [comment]
I've simplified the code in 585dc13

Basically, I gave up on counting detections on the displayed slice and
counted them on the full brain. After all, it's more meaningful.

**Pull request merged**